eng/tools/select-packages/index.js

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See the @microsoft/rush package's LICENSE file for license information.
+
+// This script searches for package.json files recursively under a given path
+// excluding packages under a node_modules folder and opens them to determine
+// whether they contain a "sdk-archetype" field with the value of "client".
+//
+// If they do, then the package name is recorded and then after all package.json files
+// have been scanned a single string is output using the Azure Pipelines Commands special
+// syntax which creates a variable for use within a later stage of the pipeine.
+var glob = require("glob");
+
+let logLineCounter = 0;
+const log = message => {
+  console.log(`select-packages(${logLineCounter++}): ${message}`);
+};
+
+log(`Working directory is "${process.cwd()}".`);
+
+let serviceDirectory = process.argv[2];
+log(`Specified service directory is "${serviceDirectory}".`);
+
+// This code assumes that in the service directory, packages that we want to publish
+// exist only in directories that are immediate children of the service directory.
+glob(`${serviceDirectory}/*/package.json`, (err, files) => {
+  if (err) {
+    log(`An error occured: ${err}`);
+    process.exit(1);
+  }
+
+  let packageTargets = "";
+
+  if (files) {
+    log(`Found ${files.length} packages under service directory.`);
+
+    for (let fileIndex in files) {
+      log(`Processing package file ${fileIndex} at "${files[fileIndex]}"`);
+      let packageContents = require(files[fileIndex]);
+
+      log(
+        `Package name for package ${fileIndex} is "${packageContents.name}".`
+      );
+
+      if (packageContents.isDataPlane) {
```

Comment:
We currently have 3 types of projects: mgmt, client (track2), and data (track1). I wonder if we should call this property sdk-type or something like that.

Reply:
Done.